tests/run-tests

```diff
@@ -552,6 +552,11 @@ the last matching regex is used:
   run-tests -i async - exclude all, then include tests containg "async" anywhere
   run-tests -e '/big.+int' - include all, then exclude by regex
   run-tests -e async -i async_foo - include all, exclude async, yet still include async_foo
+When running tests, run-tests compares the MicroPython output of the test with the output
+produced by running the test through CPython unless a <test>.exp file is found, in which
+case it is used as comparison.
+If a test fails, run-tests produces a pair of <test>.out and <test>.exp files in the current
+directory with the MicroPython output and the expectations, respectively.
```

Comment:
This reads well, but I would suggest for it to be shifted up to be added to the end of the "description" so that it's the thing people read at the start of the help text, rather than the very end.  Then it's like giving a general overview of how the script works, before going into details of the command-line arguments.